test/unittesting/utils.go

```diff
@@ -0,0 +1,227 @@
+/*
+ * Copyright (c) 2021, VRAI Labs and/or its affiliates. All rights reserved.
+ *
+ * This software is licensed under the Apache License, Version 2.0 (the
+ * "License") as published by the Apache Software Foundation.
+ *
+ * You may not use this file except in compliance with the License. You may
+ * obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+
+package unittesting
+
+import (
+	"fmt"
+	"io/ioutil"
+	"log"
+	"net/http"
+	"os"
+	"os/exec"
+	"strings"
+
+	"github.com/supertokens/supertokens-golang/recipe/emailpassword"
+	"github.com/supertokens/supertokens-golang/recipe/jwt"
+	"github.com/supertokens/supertokens-golang/recipe/openid"
+	"github.com/supertokens/supertokens-golang/recipe/passwordless"
+	"github.com/supertokens/supertokens-golang/recipe/session"
+	"github.com/supertokens/supertokens-golang/recipe/thirdparty"
+	"github.com/supertokens/supertokens-golang/recipe/thirdpartyemailpassword"
+	"github.com/supertokens/supertokens-golang/supertokens"
+)
+
+//*returns the list of all the pids inside all the running instance files inside the .started directory of the root
+func getListOfPids() []string {
+	slashesNeededToGoUp := returnNumberOfDirsToGoUpFromCurrentWorkingDir()
+	os.Setenv("INSTALL_PATH", slashesNeededToGoUp+"supertokens-root")
+	defer os.Unsetenv("INSTALL_PATH")
+	installationPath := os.Getenv("INSTALL_PATH") //*---> ../supertokens-root
+	pathOfDirToRead := installationPath + "/.started/"
+	files, err := ioutil.ReadDir(pathOfDirToRead)
+	if err != nil {
+		return []string{}
+	}
+	var result []string
+	for _, file := range files {
+		pathOfFileToBeRead := installationPath + "/.started/" + file.Name()
+		data, err := ioutil.ReadFile(pathOfFileToBeRead)
+		if err != nil {
+			log.Fatalf(err.Error())
+		}
+		result = append(result, string(data))
+	}
+	return result
+}
+
+//*this copies the config.yaml from temp folder of the root and puts it into the root level.
+func setUpST() {
+	slashesNeededToGoUp := returnNumberOfDirsToGoUpFromCurrentWorkingDir()
+	os.Setenv("INSTALL_PATH", slashesNeededToGoUp+"supertokens-root")
```

Comment:
Don't set `INSTALL_PATH` as an env var. Make the user pass it and then you figure out how many `../` needs to be prepended to it.